src/main/java/org/jfrog/hudson/pipeline/common/types/deployers/Deployer.java

```diff
@@ -171,15 +178,21 @@ public void deployArtifacts(BuildInfo buildInfo) {
         cpsScript.invokeMethod("deployArtifacts", stepVariables);
     }
 
-    public void deployArtifacts(BuildInfo buildInfo, TaskListener listener, FilePath ws) throws IOException, InterruptedException {
+    public void deployArtifacts(BuildInfo buildInfo, TaskListener listener, FilePath ws, Run build) throws IOException, InterruptedException {
         if (buildInfo.getDeployableArtifacts().isEmpty()) {
             listener.getLogger().println("No artifacts for deployment were found");
             return;
         }
         String agentName = Utils.getAgentName(ws);
         if (buildInfo.getAgentName().equals(agentName)) {
             org.jfrog.hudson.ArtifactoryServer artifactoryServer = Utils.prepareArtifactoryServer(null, server);
-            Credentials credentials = getDeployerCredentialsConfig().getCredentials(null);
+            Credentials credentials = getDeployerCredentialsConfig().getCredentials(build.getParent());
+            // Credentials.EMPTY_CREDENTIALS is only returned in the case where credentialsId is used.
```

Comment:
I suggest to slightly modify the comment and change it to:
*// Credentials.EMPTY_CREDENTIALS can only be returned when credentialsId is used.*

Reply:
@eyalbe4, The reason why i added this test, was because i got "burned" multiple times trying to reference a credentialId that either wasn't available in the context i ended up running the build job in (build was moved/reorganized), by mistake referencing a ssh key instead of a username/password.  I even debugged a case where the credentialId had a leading space in Jenkins.
So I really like the fact that it fails, when no credentials was actually returned.

However, I have been working on other types of build pipelines as of late, which are using  UploadSpec to deploy/upload their artifacts.  Lo and behold.  I have been debugging the same issues.

Currently the [`credentialsLookup()`](https://github.com/reenberg/jenkins-artifactory-plugin/blob/4272760c190f7768245f04256bbd34507375ba8d/src/main/java/org/jfrog/hudson/util/plugins/PluginsUtils.java#L59) method returns `Credentials.EMPTY_CREDENTIALS`, however I would really like to the test I implemented here inside there, and then throw the Exception in there.  If not handled there, then every use of this method would need to implement such a check, which seems errorprone.
I don't see that you would ever want to get a set of empty credential (which ultimately leads to anonymous), when you have explicitly specified a credentialId in your pipeline? 

If you agree, I would like to submit a PR one of the days